### PR TITLE
Support mistral interleaved attn

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Allow mistral models with interleaved attention patterns to be run-able in vLLM. For now we'll have to cap the max model length to the minimum of the allowed attention pattern until we figure out how to cleanly support interleaved attention pattern (similar to Gemma)

### CODE REVIEW

File: vllm/config.py [hunk 1]
```diff
@@ -422,7 +428,8 @@ def verify_with_parallel_config(
                                "pipeline parallelism currently. Disabling it.")
                 self.use_async_output_proc = False
 
-    def get_hf_config_sliding_window(self) -> Optional[int]:
+    def get_hf_config_sliding_window(
+            self) -> Optional[Union[int, List[Optional[int]]]]:
```
Comment (DarkLight1337) [inline vllm/config.py:432]:
```suggestion
            self) -> Union[Optional[int], List[Optional[int]]]]:
```

and so on for the code below, to be more readable.

File: vllm/config.py [hunk 2]
```diff
@@ -1803,6 +1813,14 @@ def _get_and_verify_max_len(
     return int(max_model_len)
 
 
+def get_min_sliding_window(
+        sliding_window: Union[int, List[Optional[int]]]) -> int:
+    if isinstance(sliding_window, list):
+        return min([s for s in sliding_window if s is not None])
```
Comment (DarkLight1337) [inline vllm/config.py:1819]:
```suggestion
        return min(s for s in sliding_window if s is not None)
```

No need to iterate twice here.

File: vllm/config.py [hunk 3]
```diff
@@ -1803,6 +1813,14 @@ def _get_and_verify_max_len(
...
+    else:
+        return sliding_window
```
Comment (DarkLight1337) [inline vllm/config.py:1821]:
```suggestion
    
    return sliding_window
```

Unnecessary else statement

**Review by DarkLight1337 — APPROVED** [review]
Overall looks good, just some nits

**Review by mgoin — APPROVED** [review]

**Comment by yananchen1989** [comment]
May I know what is the name used in vllm for Ministral ?
I only find `mistralai/Ministral-8B-Instruct-2410` on the huggingface mistral collection `https://huggingface.co/mistralai`.

Also, when I try to load it, 

> OSError: Can't load tokenizer for 'mistralai/Ministral-8B-Instruct-2410'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure 'mistralai/Ministral-8B-Instruct-2410' is the correct path to a directory containing all relevant files for a LlamaTokenizerFast tokenizer.

**Comment by mgoin** [comment]
@yananchen1989 make sure to load that model with `tokenizer_mode="mistral"` since it is a mistral format checkpoint

**Comment by yananchen1989** [comment]
`tokenizer_mode= "mistral" if args.llm_name.startswith('mistralai') else 'auto'` not sure if this writting is fine .

**Comment by patrickvonplaten** [comment]
> > @yananchen1989 make sure to load that model with `tokenizer_mode="mistral"` since it is a mistral format checkpoint
> 
> shoule I set `tokenizer_mode="mistral"` each time when I am using mistral series ？ thanks.

Yes that would be nice! We could consider making it the default actually for the mistral organization

**Comment by heheda12345** [comment]
I find the huggingface config of `mistralai/Ministral-8B-Instruct-2410` is like:
```python
"sliding_window": 32768,
```

So why do we need to support sliding window as a list?

**Pull request merged**